### PR TITLE
secrets: rework URL comments for consistency, and update vault to use environment variables for the default opener

### PR DESCRIPTION
**Pull request by vangent**
Updates #1174.

### CODE REVIEW

File: secrets/gcpkms/kms.go [hunk 1]
```diff
@@ -17,14 +17,13 @@
...
 // URLs
 //
-// For secrets.OpenKeeper URLs, gcpkms registers for the scheme "gcpkms".
-// The host+path are used as the key resource ID; see
-// https://cloud.google.com/kms/docs/object-hierarchy#key for more details.
-// Example: "gcpkms://projects/[PROJECT_ID]/locations/[LOCATION]/keyRings/[KEY_RING]/cryptoKeys/[KEY]".
-//
-// secrets.OpenKeeper will use Application Default Credentials, as described in
+// For secrets.OpenKeeper, gcpkms registers for the scheme "gcpkms".
+// The default URL opener will creating a connection using use default
```
Comment (shantuo) [inline secrets/gcpkms/kms.go]: nit: s/creating/create

File: secrets/vault/vault.go [hunk 2]
```diff
@@ -18,13 +18,12 @@
...
-//   - address: Sets Config.APIConfig.Address; should be a full URL with the
-//       address of the Vault server.
-//   - token: Sets Config.Token; the access token the Vault client will use.
-// Example URL: "vault://mykey?address=http://vault.server.com:8080&token=aaaaa".
+// For secrets.OpenKeeper, vault registers for the scheme "vault".
+// The default URL opener will dial a Vault server using the environment
+// variables "VAULT_SERVER_URL" and "VAULT_SERVER_TOKEN".
+// To customize the URL opener, or for more details on the URL format,
+// see URLOpener.
+// See https://godoc.org/gocloud.dev#URLs for background information.
```
Comment (shantuo) [inline secrets/vault/vault.go]: s/`https://godoc.org/gocloud.dev#URLs`/`https://godoc.org/gocloud.dev#hdr-URLs`

And also for comments in other files.

File: secrets/gcpkms/kms.go (hunk 1 above)
Comment (vangent) [reply secrets/gcpkms/kms.go]: Done.

File: secrets/vault/vault.go (hunk 2 above)
Comment (vangent) [reply secrets/vault/vault.go]: Nice catch, done. Also fixing the As links.

**Pull request merged**